### PR TITLE
Dark theme toggle

### PR DESCRIPTION
**Pull request by rogerhutchings**
Package: app-project

This PR: 

- Standardises use of the theme `{ dark: true }` prop, which is how Grommet determines dark theme usage
- Removes some styled-theming stuff
- Adds a toggle for dark theme
- Moves theme merging out to a separate component in order to observe the store
- Fixes some tests
- Lint
- Adds the Project Name

Still to do:

- Some modals aren't picking up the theme object properly, so don't toggle colour
- Add icons to toggle button
- Update dark theme colours on LCV
- Style some of the `Box` elevations for dark mode
- Set current mode in a cookie so we can render dark mode server-side
- Make project name, theme toggle sticky
- Some buttons are coming out green

...but I thought this PR was getting too big already

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Looks good but something I've noticed after a quick look running the app locally: each new page switches back to the light theme when it loads.

**Comment by eatyourgreens** [comment]
Design comment: people are probably going to miss the toggle on the right. For me, it's half-hidden behind the browser scrollbar.

**Comment by rogerhutchings** [comment]
I've added another todo for storing a cookie for the mode, so we can render it on the server. @beckyrother do you mind me adding more padding to the toggle so it can be seen past the scrollbars?

File: packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/FieldGuide.js [hunk 1]
```diff
@@ -1,35 +1,50 @@
...
   }
 }
 
+@withTheme
 @inject(storeMapper)
 @observer
 class FieldGuide extends React.Component {
   render () {
     const { activeItemIndex, className, items } = this.props
+    console.info(this.props.theme)
```
Comment (shaunanoordin) [inline packages/lib-classifier/src/components/Classifier/components/FieldGuide/components/FieldGuide/FieldGuide.js]: (minor) Is this unnecessary? (Strangely, I can't seem to trigger this line. 🤔 )

**Comment by shaunanoordin** [comment]
Query: is the Field Guide supposed to reflect the dark theme too? I see that `/src/components/Classifier/components/FieldGuide/components/FieldGuide/FieldGuide.js` was modified, but I'm not sure if this is intended:

Screen Shot 2019-04-17 at 18 06 35
_White-background Field Guide on a dark-themed Classifier page, on app-project_

**Comment by beckyrother** [comment]
@rogerhutchings yes, please add padding so the scroll bar doesn't cover up these buttons.

@shaunanoordin yes, the modals should all be the dark theme as well.

**Comment by rogerhutchings** [comment]
@shaunanoordin yup, for some reason the modals aren't picking up the theme change. it's in my todos in the PR, and i'll create these as separate issues.

**Comment by rogerhutchings** [comment]
@shaunanoordin @eatyourgreens is this okay to merge?

**Comment by beckyrother** [comment]
What's the link to look at it before merging?

**Pull request merged**